src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -322,13 +322,31 @@ private static string GenerateUpgradeReport(NuGetProject nuGetProject, string ba
             bool acceptedLicense = true;
 
             List<string> removedPackages = null;
+            List<Tuple<string, string>> existingPackages = new List<Tuple<string, string>>();
             List<Tuple<string,string>> addedPackages = null;
             List<Tuple<string,string>> updatedPackagesOld = null;
             List<Tuple<string,string>> updatedPackagesNew = null;
 
             // Enable granular level telemetry events for nuget ui operation
             uiService.ProjectContext.OperationId = Guid.NewGuid();
 
+            try
+            {
+                // collect the install state of the existing packages
+                foreach (var project in uiService.Projects)
```

Comment:
I think that when on the solution package manager UI, this would call GetInstalledPackages on every project in the solution. 
This can be very expensive. Not in a, it will cause NuGet sluggishness cause of UI Delays, but these can be expensive operations.   

This might have already been an issue with the previous telemetry though. So I'd just try to verify how the telemetry behaves when on the Solution PM UI.

Reply:
In my testing, it only seems to call it on the projects _selected_ projects.  It does not call it for unchecked projects.  I think that's the correct behavior.

Interestingly, one thing that does happen is that the listed of existing packages is not unique; if two different projects have some of the same nugets, then they will appear multiple times in the existingpackages property.

That feels wrong.  What do you think?

Reply:
> That feels wrong. 

I agree. It's not very useful. In the worst case it's misleading.
But that's a problem with the specific package telemetry in general. We should fix though :)